### PR TITLE
Update github repo links

### PR DESCRIPTION
**Pull request by lewiszlw**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion/issues/9691

### CODE REVIEW

File: docs/source/user-guide/introduction.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ Here are some active projects using DataFusion:
 
 - [Arroyo](https://github.com/ArroyoSystems/arroyo) Distributed stream processing engine in Rust
 - [Ballista](https://github.com/apache/arrow-ballista) Distributed SQL Query Engine
```
Comment (Jefffrey) [inline docs/source/user-guide/introduction.md:98]: I guess we can tackle ballista update in a separate PR?

**Review by Jefffrey — APPROVED** [review]
Nice :+1: 

(I tried to look at every line; those changelog diffs are big O.O)

**Comment by alamb** [comment]
I merged up from main to resolve a conflict with this PR. I think it has a high liklihood of conflicts so I think we should merge this PR ASAP

**Review by comphead — COMMENTED** [review]
wow, epic work @lewiszlw I clicked on some links, it works, but Installation link gives me 404

**Comment by comphead** [comment]
> wow, epic work @lewiszlw I clicked on some links, it works, but Installation link gives me 404

But we can do that in follow up PR to avoid conflicts

**Review by andygrove — APPROVED** [review]

**Comment by andygrove** [comment]
Thanks @lewiszlw

**Pull request merged**